worker/xml/xml_generator.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import os
+import json
+
+from jinja2 import Template
+
+log = logging.getLogger(__name__)
+
+
+def generate_xml(work_path, data=None, xml_type='marc'):
+    """
+    Main function for XML-Generation. Uses json file as parameters for the
+    XML template and writes the generated files to the working directory.
+    :param work_path: Directory of the job
+    :param data: XML metadata
+    :param xml_type: used to choose the template file - either marc or ojs
+    :return: None
+    """
+    data = data or _load_json_data(work_path)
+    template = _read_template_file(xml_type)
+    if xml_type == 'marc':
+        prepared_data = _prepare_marc_data(data)
+        for article in prepared_data:
+            filled_template = template.render(article)
+            _write_template_to_file(filled_template,
+                                    work_path,
+                                    article['title'].replace(' ', '_') + ".xml")
+    else:
+        prepared_data = _prepare_ojs_data(data)
+        filled_template = template.render(prepared_data)
+        _write_template_to_file(filled_template, work_path, 'ojs_import.xml')
+
+
+def _load_json_data(work_path):
+    json_path = os.path.join(work_path, 'data_json/data.json')
+    with open(json_path) as data_object:
+        data = json.load(data_object)
+    return data
+
+
+def _read_template_file(template_type='marc'):
+    path = "resources/" + template_type + "_template.xml"
+    template_file = open(path, 'r')
+    template = Template(template_file.read(), trim_blocks=True, lstrip_blocks=True)
+    template_file.close()
+
+    return template
+
+
+def _write_template_to_file(template, target_path, target_file_name):
```

Comment:
Würde ich umbenennen in `_write_xml_to_file`. Es wird ja nicht das Template geschrieben.